### PR TITLE
Add magit-repolist support

### LINKED ISSUE #7083: Magit Feature Request `magit-list-repositories`
I think it would be nice to have `<space>gL` binding for `magit-list-repositories`. It's trivial to add the binding, but where I got hung up is the buffer only responds to `emacs` mode bindings (ctrl-z then a return will open the status buffer). I'm not sure how to fix it.

In my `dotspacemacs/user-config()` I configured the feature with:

``` lisp
  (setq magit-repository-directories 
      '( "~/Development" ))
```

### PR DESCRIPTION
**Pull request by rphillips**
Adds Magit Repolist support.

Bindings:

`SPC gL`
`RET`
`g`

Fixes #7083

### CODE REVIEW

File: layers/+source-control/git/packages.el [hunk 1]
```diff
@@ -154,6 +155,11 @@ Press [_b_] again to blame further in the history, [_q_] to go up or quit."
       (require 'git-rebase)
       ;; bind function keys
       ;; (define-key magit-mode-map (kbd "<tab>") 'magit-section-toggle)
+      (evilified-state-evilify-map magit-repolist-mode-map
+          :mode magit-repolist-mode
+          :bindings
+          (kbd "g") 'magit-list-repositories
```
Comment (nixmaniack) [inline layers/+source-control/git/packages.el]: I think it should be bound to `gr` as we do in other buffers to _refresh_ the view. Binding it to `g` would override some evil bindings such `gg` to goto the top of buffer.

**Comment by deb0ch** [comment]
Just tried to call the function `magit-list-repositories` and it just gave an empty page.

I guess it is because I did not set the options `magit-repository-directories' and `magit-repository-directories-depth'.

Would there be a nice default value to set that to ? like try to look in the entire home directory ? Or is it a bad idea ?

Maybe it would also be nice to bind `q` and `ESC` like in the other magit buffers ?

File: layers/+source-control/git/packages.el (hunk 1 above)
Comment (rphillips) [inline layers/+source-control/git/packages.el]: good call. tweaked in 386b0b1

**Comment by rphillips** [comment]
@deb0ch `q` is bound already. I worry about a default directory, because I have _a lot_ of top level projects. It would be nice if someone knew more lisp, to display a warning if the directories are not set.

**Comment by TheBB** [comment]
Thanks! Cherry-picked in develop. I made some small changes to the documentation. You can safely delete your branch.

**Pull request closed without merging**